frontend/src/components/Payment.tsx

```
@@ -11,280 +11,293 @@ import { TxStatus } from "../types/Transaction";
+        if (typeof err === "object" && err !== null) {
+            // Convert to string if it's an Error object
+            const errorString = err.message || JSON.stringify(err);
 
-        const disperse = new ethers.Contract(
-          disperseAddress,
-          Disperse.abi,
-          signer
-        );
+            // Try to match the specific error format
+            const match = errorString.match(/Details: (execution reverted: [^\\]*)/);
+            if (match && match[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)